src/App/Tracker/Controller/Hooks/Listeners/AbstractListener.php

```diff
@@ -325,4 +321,117 @@ protected function createStatus(Github $gitHub, TrackerProject $project, $issueN
 			$status->state, $status->targetUrl, $status->description, $status->context
 		);
 	}
+
+	/**
+	 * Set Categories
+	 *
+	 * @param   object       $hookData       Hook data payload
+	 * @param   Logger       $logger         Logger object
+	 * @param   object       $project        Object containing project data
+	 * @param   IssuesTable  $table          Table object
+	 * @param   array        $addCategories  The Categories to add
+	 *
+	 * @return  void
+	 *
+	 * @since   1.0
+	 * @throws  \RuntimeException
+	 */
+	protected function setCategories($hookData, Logger $logger, $project, IssuesTable $table, $addCategories)
+	{
+		// The Github ID if we have a pull or issue so that method can handle both
+		$issueNumber = $this->getIssueNumber($hookData);
+
+		if ($issueNumber === null)
+		{
+			$message = sprintf('Error retrieving issue number for %s/%s', $project->gh_user, $project->gh_project);
+
+			$logger->error($message);
+
+			throw new \RuntimeException($message);
+		}
+
+		$categoryModel            = new CategoryModel($this->getContainer()->get('db'));
```

Comment:
They were unused.  If you want to inject the container in you'll have to
add those back and make sure setContainer is called when creating the
listener

On Saturday, April 2, 2016, zero-24 <notifications@github.com> wrote:

> In src/App/Tracker/Controller/Hooks/Listeners/AbstractListener.php
> <https://github.com/joomla/jissues/pull/770#discussion_r58299300>:
>
> > +	 */
> > +	protected function setCategories($hookData, Logger $logger, $project, IssuesTable $table, $addCategories)
> > +	{
> > +		// The Github ID if we have a pull or issue so that method can handle both
> > +		$issueNumber = $this->getIssueNumber($hookData);
> > +
> > +		if ($issueNumber === null)
> > +		{
> > +			$message = sprintf('Error retrieving issue number for %s/%s', $project->gh_user, $project->gh_project);
> > +
> > +			$logger->error($message);
> > +
> > +			throw new \RuntimeException($message);
> > +		}
> > +
> > +		$categoryModel            = new CategoryModel($this->getContainer()->get('db'));
>
> hmm maybe i have removed one use to much? Here: b1d485d
> <https://github.com/joomla/jissues/pull/770/commits/b1d485d6db82ade99ec5ca3df7a64eacdb937fd3>
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/joomla/jissues/pull/770/files/b1d485d6db82ade99ec5ca3df7a64eacdb937fd3#r58299300>
>